sympy/core/traversal.py

```diff
@@ -27,6 +29,8 @@ def iterargs(expr):
     args = [expr]
     for i in args:
         yield i
+        if not isinstance(i.args, Iterable):
+            continue
```

Comment:
I'm not sure about this patch. All SymPy basic objects expect `args` to be tuples

Reply:
The error is caused due to the using of Function which is not an object Ex: Derivative(f, x). Which returns property (not iterable) causing error.